crates/polars-parquet/src/arrow/read/deserialize/binview/basic.rs

```diff
@@ -102,33 +105,78 @@ impl<'a> utils::Decoder<'a> for BinViewDecoder {
             BinaryState::OptionalDictionary(page_validity, page_values) => {
                 // Already done on the dict.
                 validate_utf8 = false;
+
                 let page_dict = &page_values.dict;
+                let offsets = page_dict.offsets();
+
+                // @NOTE: If there is no lengths (i.e. 0-1 offset), then we will have only nulls.
+                let max_length = offsets.lengths().max().unwrap_or(0);
+
+                // We do not have to push the buffer if all elements fit as inline views.
+                let buffer_idx = if max_length <= View::MAX_INLINE_SIZE as usize {
+                    0
+                } else {
+                    values.push_buffer(page_dict.values().clone())
```

Comment:
Ah, nice. So we don't have to go through the builder, but immediately use the buffer. :+1: